internal/archive/archive.go

```diff
@@ -129,6 +138,44 @@ func (a *ubuntuArchive) Fetch(pkg string) (io.ReadCloser, error) {
 const ubuntuURL = "http://archive.ubuntu.com/ubuntu/"
 const ubuntuPortsURL = "http://ports.ubuntu.com/ubuntu-ports/"
 
+var proArchiveInfo = map[string]struct {
+	baseURL, label string
+}{
+	"fips": {
+		baseURL: "https://esm.ubuntu.com/fips/ubuntu/",
+		label:   "UbuntuFIPS",
+	},
+	"fips-updates": {
+		baseURL: "https://esm.ubuntu.com/fips-updates/ubuntu/",
+		label:   "UbuntuFIPSUpdates",
+	},
+	"apps": {
+		baseURL: "https://esm.ubuntu.com/apps/ubuntu/",
+		label:   "UbuntuESMApps",
+	},
+	"infra": {
+		baseURL: "https://esm.ubuntu.com/infra/ubuntu/",
+		label:   "UbuntuESM",
+	},
+}
+
+func archiveURL(pro, arch string) string {
```

Comment:
Can you document the function saying that we expect the pro value to be validated and same for openUbuntu.

Reply:
Done.

Reply:
We can just check the map in `archive.Open` and return an error if the pro value is not valid. Then drop the comment here about being validated because this is an internal function and the external one is doing the work.

Reply:
Done in 2608ee3ab302dc5dde18995c7f437e2bbc44ed21. Thanks.